### ISSUE TITLE
Fatal Exception: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState

### ISSUE THREAD

**Issue by katajona** [Iterable/iterable-android-sdk#322]
I have a crash for some of my users, recorded by crashlytics in version `3.2.10`:

```
Fatal Exception: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
       at androidx.fragment.app.FragmentManager.checkStateLoss(FragmentManager.java:1703)
       at androidx.fragment.app.FragmentManager.enqueueAction(FragmentManager.java:1743)
       at androidx.fragment.app.BackStackRecord.commitInternal(BackStackRecord.java:321)
       at androidx.fragment.app.BackStackRecord.commit(BackStackRecord.java:286)
       at androidx.fragment.app.DialogFragment.dismissInternal(DialogFragment.java:276)
       at androidx.fragment.app.DialogFragment.dismiss(DialogFragment.java:228)
       at com.iterable.iterableapi.IterableInAppFragmentHTMLNotification.access$601(IterableInAppFragmentHTMLNotification.java:37)
       at com.iterable.iterableapi.IterableInAppFragmentHTMLNotification$5.run(IterableInAppFragmentHTMLNotification.java:386)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1057)
       at android.view.Choreographer.doCallbacks(Choreographer.java:875)
       at android.view.Choreographer.doFrame(Choreographer.java:772)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1042)
       at android.os.Handler.handleCallback(Handler.java:888)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:213)
       at android.app.ActivityThread.main(ActivityThread.java:8178)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1101)
```

**Comment by Ayyanchira** [comment]
Thanks for reporting this @katajona. This is helpful.
Will take a look on why thats happening.

**Comment by Ayyanchira** [comment]
Hi @Katajona, is/are there any specific device brands or type in which the crash is occurring? Is it reproducible? Any details can be helpful.
Thanks.

**Comment by katajona** [comment]
Nothing specific, 86% Huawei, 7% Samsung 4% Sony 3% Other. Based on Android phone version Android 10: 61%, Android 9: 32%, Android 10: 61%. Proximity on is 100%. We are showing an InApp campaign right after app update. So far the crash happened for 85 users. I am using `IterableConfig.Builder().setLogLevel(Log.VERBOSE).setUrlHandler { uri, actionContext -> //Handling deep link}`
So far I could not reproduce it. I think you are trying to show the in-app right when the user is closing the app/changing the activities.

**Comment by katajona** [comment]
@Ayyanchira When will this be released? It already caused 512 crashes.

**Comment by Ayyanchira** [comment]
It will be released early next week.

**Comment by Ayyanchira** [comment]
@katajona , we have released 3.2.11 with the fix. Hope your users wont see those crashes :)
